python/plugins/processing/algs/gdal/ClipRasterByExtent.py

```diff
@@ -161,7 +162,32 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
             extra = self.parameterAsString(parameters, self.EXTRA, context)
             arguments.append(extra)
 
-        arguments.append(inLayer.source())
+        layerSource = inLayer.source()
+        layerProvider = inLayer.dataProvider()
+        if (layerProvider.name() == 'postgresraster'):
```

Comment:
Can you move this to GdalUtils to avoid the duplicate code?

Reply:
Good point @nyalldawson. Thanks.